metricshub-doc/src/site/markdown/connector-design-guide/connector-object-references.md

```diff
@@ -0,0 +1,171 @@
+keywords: guide, connector, connector, object, reference
+description: This page shows how to reference connector objects such as sources, entries, columns and files, etc. inside the YAML document.
+
+# Connector Object References
+
+This page shows how to reference connector objects such as sources, entries, columns and files, etc. inside the YAML document.
+
+* [Source Reference](#source_reference)
+    * [Format](#source-format)
+    * [Example](#source-example)
+* [Entry Reference](#entry_reference)
+    * [Format](#entry-format)
+    * [Example](#entry-example)
+* [Column Reference](#column_reference)
+    * [Format](#column-format)
+    * [Example](#column-example)
+* [File Reference](#file_reference)
+    * [Format](#file-format)
+    * [Example](#file-example)
+* [Mono-Instance Reference](#mono-instance_reference)
+    * [Format](#mono-instance-format)
+    * [Example](#mono-instance-example)
+* [Translation Table Reference](#translation_table_reference)
+    * [Format](#translation-table-format)
+    * [Example](#translation-table-example)
+* [Awk Script Reference](#awk_script_reference)
+    * [Format](#awk-script-format)
+    * [Example](#awk-script-example)
+
+## Source Reference
+
+### <a id="source-format" />Format
+
+A Source can be referenced either with its full path or its relative path. A relative path can be used only if the referenced source is in the same monitor and same job as the current source.
+
+### <a id="source-example" />Example
+```yaml
+      mapping:
+        # InstanceTable = Source(1)
+        source: ${source::monitors.disk_controller.discovery.sources.source_discovery}
+```
+
+```yaml
+        sourceDiscovery:
+          type: tableJoin
+          leftTable: ${source::monitors.enclosure.collect.sources.source_chassis} # full path for a source in another monitor
+          rightTable: ${source::source_enclosure} # relative path
+          leftKeyColumn: 1
+          rightKeyColumn: 1
+```
+
+## Entry Reference
+
+### <a id="entry-format" />Format
+```yaml
+$<columnNumber>
+```
+
+### <a id="entry-example" />Example
+```yaml
+        source(3):
+          type: http
+          method: GET
+          executeForEachEntryOf:
+            source: ${source::monitors.enclosure.discovery.sources.source(2)}
+            concatMethod: list
+          url: /api/rest/StorageCenter/ScChassis/$2/PowerSupplyList

```

Comment:
`path: /api/rest/StorageCenter/ScChassis/$2/PowerSupplyList`

Reply:
Why path instead of url ? This example is taken directly from a connector.

Reply:
All the `URL` field have been renamed to `path`. You can refer to #5 for more information.